usim_pytest/test_types/test_pipe.py

```diff
@@ -84,3 +84,41 @@ class TestPipe:
         async with Scope() as scope:
             for _ in range(6):
                 scope.do(pipe.transfer(total=15))
+
+
+class TestUnboundedPipe:
+    @assertion_mode
+    @via_usim
+    async def test_debug_misuse(self):
+        with pytest.raises(AssertionError):
+            UnboundedPipe(throughput=0)
+        with pytest.raises(AssertionError):
+            UnboundedPipe(throughput=-10)
+        with pytest.raises(AssertionError):
+            UnboundedPipe(throughput=10)
+        pipe = UnboundedPipe()
+        with pytest.raises(AssertionError):
+            await pipe.transfer(total=-10, throughput=20)
+        with pytest.raises(AssertionError):
+            await pipe.transfer(total=10, throughput=0)
+
+    @via_usim
+    async def test_transfers(self):
+        pipe = UnboundedPipe()
+        for total in (0, 10, float('inf')):
+            await pipe.transfer(total=total)
+        assert (time == 0)
+        async with Scope() as scope:
+            for total in (0, 10, float('inf')):
+                scope.do(pipe.transfer(total=total))
+                scope.do(pipe.transfer(total=total))
+                scope.do(pipe.transfer(total=total))
+                scope.do(pipe.transfer(total=total))
+        assert (time == 0)
```

Comment:
Could you please add a test here with ``transfer > 0`` and ``throughput > 0 and throughput < transfer``?
Thanks :)

Reply:
Ah, that's a good point. Totally missed that one.